examples/cross-contract-high-level/src/lib.rs

```diff
@@ -20,12 +15,8 @@ pub trait ExtCrossContract {
     fn merge_sort(&self, arr: Vec<u8>) -> PromiseOrValue<Vec<u8>>;
     fn merge(
         &self,
-        #[callback_unwrap]
-        #[serializer(borsh)]
-        data0: Vec<u8>,
-        #[callback_unwrap]
-        #[serializer(borsh)]
-        data1: Vec<u8>,
```

Comment:
That was our thought too.  It also doesn't have parity with the other example since it uses borsh for the external interface.  We had initial rewritten to use a JSON method that then called the inner merge, but found that to be way to confusing.

These two examples in general try to combine too many concepts: Factory pattern, cross contract calls, callback_unwrap, and borsh.  They should be combined into one project that provides examples for each at higher and lower levels.  https://github.com/near-examples/cross-contract-calls tries to do this for assemblyscript with sim tests included.

Reply:
related https://github.com/near/near-sdk-rs/issues/589 https://github.com/near/near-sdk-rs/issues/590